.github/workflows/java-cd.yml

```
@@ -158,3 +137,41 @@ jobs:
                   name: java-${{ matrix.host.TARGET }}
                   path: |
                       java/bundle*.jar
+
+    combine-binaries-to-one-directory:
+        needs: create-binaries-to-publish
+        runs-on: ubuntu-latest
+        steps:
+            - name: Download published artifacts
+              uses: actions/download-artifact@v4
+
+            - name: Set the release version
+              shell: bash
+              run: |
+                if ${{ github.event_name == 'pull_request' }}; then
+                  R_VERSION="255.255.255"
+                elif ${{ github.event_name == 'workflow_dispatch' }}; then
+                  R_VERSION="${{ env.INPUT_VERSION }}"
+                else
+                  R_VERSION=${GITHUB_REF:11}
+                fi
+                echo "RELEASE_VERSION=${R_VERSION}" >> $GITHUB_ENV
+                echo "Release version detected: $R_VERSION"
+              env:
+                INPUT_VERSION: ${{ github.event.inputs.version }}
+
+            - name: Move all files to one zip file
+            # TODO: Add `./java-aarch64-unknown-linux-gnu/* bundles` when we are able to support a linux arm runner.
+            # TODO: Add "maven" to linux arm PACKAGE_MANAGER in build-matrix.json.
+              run: |
+                mkdir bundles
+                mv ./java-aarch64-apple-darwin/* bundles
+                mv ./java-x86_64-apple-darwin/* bundles
+                mv ./java-x86_64-unknown-linux-gnu/* bundles
```

```suggestion
                mv ./java-*/* bundles
```
This will do the same and does not require change if you add new platform(s)